projects/cashmere/src/lib/datepicker/calendar/calendar.component.ts

```diff
@@ -532,7 +532,14 @@ export class CalendarComponent implements AfterContentInit, AfterViewChecked, On
             curHour++;
             const tempCycle = this._hourCycle > 12 ? 23 : 12;
             if (curHour > tempCycle) {
-                curHour = tempCycle;
+                curHour = this._hourCycle > 12 ? 0 : 1;
+            }
+            if (curHour === 12) {
+                if (this._period.value === 'am') {
```

Comment:
Pretty minor, but looks like this whole conditional for `am`, `pm` switch is something that could probably be pulled into a single method

Reply:
Or instead to make it more simple, just a ternary (assuming that it's always going to be `am` or `pm`? I'm not sure if there's a 24-hour setup for this

Reply:
@jonest A separate method is what I was also thinking too. I can try a few things with the ternary but if that doesn't work, would you have an alternative idea?